unit/workflow/function-agent.test.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  AgentToolCall,
+  AgentToolCallResult,
+  FunctionAgent,
+  FunctionTool,
+  type ChatMessage,
+} from "llamaindex";
+import { describe, expect, test } from "vitest";
+import { z } from "zod";
+import { setupMockLLM } from "./llm.js";
+
+describe("FunctionAgent", () => {
```

Comment:
is it testing public methods handleToolCallResults, takeStep and finalize?
why is it testing private methods getToolCallFromResponseChunk?